src/components/CalendarDay.jsx

```diff
@@ -48,6 +55,9 @@ export default class CalendarDay extends React.Component {
         onClick={e => this.onDayClick(day, e)}
       >
         <span className="CalendarDay__day">{day.format('D')}</span>
+        {details && (
+          <span className="CalendarDay__details">{details}</span>
```

Comment:
You could probably simplify this a bit and just have this be 
```
<span className="CalendarDay__details">{renderDetails(day)}</span>
```
and deconstruct `renderDetails` from the props. I'm not sure that there's a need for the function or to separate out the variable.